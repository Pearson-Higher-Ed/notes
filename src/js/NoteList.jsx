import React, {Component} from 'react';
import Note from './Note';
import ReactDOM from 'react-dom';

export default class NoteList extends Component {
  constructor(props) {
    super(props);

    this.state = {notes: this.props.notes};
    this.removeNote = this.removeNote.bind(this);
  }

  removeNote(noteId) {
    if (this.props.store) {
      this.props.store.dispatch(this.props.actions.deleteAnnotation(noteId));
    }
    
    const removeNoteIndex = this.state.notes.findIndex(note => note.id === noteId);
    this.state.notes.splice(removeNoteIndex, 1);
    this.setState({notes: this.state.notes});
  }

  renderEmpty() {
    return (
      <div className="empty-help" >
         <div className="empty-message" tabIndex="0">
             <p>When you highlight text or add notes, they will appear here.</p>
         </div>
      </div>
    )
  }

  handleFocus(id) {
    let Note = this.refs['Note' + id];
    if (!Note) return;
    let a = Note.refs.a;
    ReactDOM.findDOMNode(a).focus();
  }

  renderNotes() {
    const that = this;

    return this.props.notes.map(function(note, i) {
      return (
        <Note key={i}
              id={i}
              noteId={note.id}
              author={note.author}
              pageId={note.pageId}
              color={note.color}
              comment={note.comment}
              text={note.text}
              time={note.time}
              removeNote={that.removeNote} 
              handleFocus={that.handleFocus.bind(that)} 
              ref={'Note' + i}/>
      );
    });
  }

  render() {
    return (
      <div id="notes" role="main">
          <div className="notes-body">
              {(this.props.notes.length === 0) ? this.renderEmpty() : this.renderNotes()}
          </div>
          <div id="notes-assert-container" role="alert" aria-live="assertive" className="reader-only"></div>
      </div>
    )
  }
}
