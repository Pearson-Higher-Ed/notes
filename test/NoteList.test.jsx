/**
PEARSON PROPRIETARY AND CONFIDENTIAL INFORMATION SUBJECT TO NDA
 *  Copyright Â© 2017 Pearson Education, Inc.
 *  All Rights Reserved.
 *
 * NOTICE:  All information contained herein is, and remains
 * the property of Pearson Education, Inc.  The intellectual and technical concepts contained
 * herein are proprietary to Pearson Education, Inc. and may be covered by U.S. and Foreign Patents,
 * patent applications, and are protected by trade secret or copyright law.
 * Dissemination of this information, reproduction of this material, and copying or distribution of this software
 * is strictly forbidden unless prior written permission is obtained
 * from Pearson Education, Inc.
**/

/* eslint-disable */

import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider } from 'react-intl';
import { InternationalSupport } from '@pearson-incubator/aquila-js-core';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'
import NoteList from '../src/js/NoteList';
import msgObject from '../translations';

const intlObj = new InternationalSupport(msgObject, 'en');

const clickNoteHandler = () => {
   console.log('clickBookmarkHandler');
 };

 const length = 5;

	it('renders correctly', () => {
	  const tree = renderer
	    .create(<MuiThemeProvider><IntlProvider locale={intlObj.getLocale()} messages={intlObj.getMessages()}><NoteList
	    	notes={[{test:'test'},{test:'test'},{test:'test'}]}
        clickNoteHandler={function(){}}
        removeNoteHandler={function(){}}
       ></NoteList></IntlProvider></MuiThemeProvider>)
	    .toJSON();
	  expect(tree).toMatchSnapshot();
	});

	it('renders the Note List', () => {
  	const component = renderer
	    .create(<MuiThemeProvider><IntlProvider locale={intlObj.getLocale()} messages={intlObj.getMessages()}><NoteList
	    	notes={[{test:'test'},{test:'test'},{test:'test'}]}
        clickNoteHandler={clickNoteHandler}
        removeNoteHandler={function(){}}
        length={length}
       ></NoteList></IntlProvider></MuiThemeProvider>);
	    let tree = component.toJSON();
	    console.log(tree);
	    tree.children[0].children[0].children[0].props.onFocus({preventDefault(){}, target:{parentNode:{classList:{add(){}}}}});
	    tree.children[0].children[0].children[1].props.onClick({preventDefault(){}, target:{parentNode:{classList:{add(){}}}}});
	    //tree.children[0].children[0].children[2].props.onKeyUp({preventDefault(){}, which:37, keyCode:37, target:{parentNode:{classList:{remove(){}}}}});
	    // console.log(tree.children[0].children[1].children[0])
	    // console.log(tree.children[0].children[2].children[0])
  		expect(tree).toMatchSnapshot();


 // tree.children[0].props.onFocus();
  //tree.children[0].props.onClick();
  // tree.children[0].props.onKeyUp();
  // tree.children[1].props.onClick();

  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});




