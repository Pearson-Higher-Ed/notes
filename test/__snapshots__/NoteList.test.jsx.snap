// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  id="notes"
  role="main"
>
  <div
    className="notes-body"
  >
    <div
      className="note-row"
    >
      <a
        className="note-link"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyUp={[Function]}
        role="link"
        tabIndex="0"
      >
        <span
          className="undefined color-dot"
        >
          <span
            className="reader-only"
          >
             Highlight
          </span>
        </span>
        <div
          className="note-content"
        >
          <div>
            <p
              className="dotdotdot"
            >
              <strong />
            </p>
          </div>
          <div
            className="hide"
          >
            <p
              className="user-note"
            />
          </div>
          <span
            className="note-date"
          >
            <time
              value={undefined}
            >
              Invalid Date
            </time>
             
            <time
              value={undefined}
            >
              Invalid Date
            </time>
          </span>
        </div>
      </a>
      <a
        aria-label="Remove note"
        className="remove"
        onClick={[Function]}
        role="link"
        tabIndex="0"
      >
        
      </a>
    </div>
    <div
      className="note-row"
    >
      <a
        className="note-link"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyUp={[Function]}
        role="link"
        tabIndex="0"
      >
        <span
          className="undefined color-dot"
        >
          <span
            className="reader-only"
          >
             Highlight
          </span>
        </span>
        <div
          className="note-content"
        >
          <div>
            <p
              className="dotdotdot"
            >
              <strong />
            </p>
          </div>
          <div
            className="hide"
          >
            <p
              className="user-note"
            />
          </div>
          <span
            className="note-date"
          >
            <time
              value={undefined}
            >
              Invalid Date
            </time>
             
            <time
              value={undefined}
            >
              Invalid Date
            </time>
          </span>
        </div>
      </a>
      <a
        aria-label="Remove note"
        className="remove"
        onClick={[Function]}
        role="link"
        tabIndex="0"
      >
        
      </a>
    </div>
    <div
      className="note-row"
    >
      <a
        className="note-link"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyUp={[Function]}
        role="link"
        tabIndex="0"
      >
        <span
          className="undefined color-dot"
        >
          <span
            className="reader-only"
          >
             Highlight
          </span>
        </span>
        <div
          className="note-content"
        >
          <div>
            <p
              className="dotdotdot"
            >
              <strong />
            </p>
          </div>
          <div
            className="hide"
          >
            <p
              className="user-note"
            />
          </div>
          <span
            className="note-date"
          >
            <time
              value={undefined}
            >
              Invalid Date
            </time>
             
            <time
              value={undefined}
            >
              Invalid Date
            </time>
          </span>
        </div>
      </a>
      <a
        aria-label="Remove note"
        className="remove"
        onClick={[Function]}
        role="link"
        tabIndex="0"
      >
        
      </a>
    </div>
  </div>
  <div
    aria-live="assertive"
    className="reader-only"
    id="notes-assert-container"
    role="alert"
  />
</div>
`;

exports[`renders the Note List 1`] = `
<div
  id="notes"
  role="main"
>
  <div
    className="notes-body"
  >
    <div
      className="note-row"
    >
      <a
        className="note-link"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyUp={[Function]}
        role="link"
        tabIndex="0"
      >
        <span
          className="undefined color-dot"
        >
          <span
            className="reader-only"
          >
             Highlight
          </span>
        </span>
        <div
          className="note-content"
        >
          <div>
            <p
              className="dotdotdot"
            >
              <strong />
            </p>
          </div>
          <div
            className="hide"
          >
            <p
              className="user-note"
            />
          </div>
          <span
            className="note-date"
          >
            <time
              value={undefined}
            >
              Invalid Date
            </time>
             
            <time
              value={undefined}
            >
              Invalid Date
            </time>
          </span>
        </div>
      </a>
      <a
        aria-label="Remove note"
        className="remove"
        onClick={[Function]}
        role="link"
        tabIndex="0"
      >
        
      </a>
    </div>
    <div
      className="note-row"
    >
      <a
        className="note-link"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyUp={[Function]}
        role="link"
        tabIndex="0"
      >
        <span
          className="undefined color-dot"
        >
          <span
            className="reader-only"
          >
             Highlight
          </span>
        </span>
        <div
          className="note-content"
        >
          <div>
            <p
              className="dotdotdot"
            >
              <strong />
            </p>
          </div>
          <div
            className="hide"
          >
            <p
              className="user-note"
            />
          </div>
          <span
            className="note-date"
          >
            <time
              value={undefined}
            >
              Invalid Date
            </time>
             
            <time
              value={undefined}
            >
              Invalid Date
            </time>
          </span>
        </div>
      </a>
      <a
        aria-label="Remove note"
        className="remove"
        onClick={[Function]}
        role="link"
        tabIndex="0"
      >
        
      </a>
    </div>
    <div
      className="note-row"
    >
      <a
        className="note-link"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyUp={[Function]}
        role="link"
        tabIndex="0"
      >
        <span
          className="undefined color-dot"
        >
          <span
            className="reader-only"
          >
             Highlight
          </span>
        </span>
        <div
          className="note-content"
        >
          <div>
            <p
              className="dotdotdot"
            >
              <strong />
            </p>
          </div>
          <div
            className="hide"
          >
            <p
              className="user-note"
            />
          </div>
          <span
            className="note-date"
          >
            <time
              value={undefined}
            >
              Invalid Date
            </time>
             
            <time
              value={undefined}
            >
              Invalid Date
            </time>
          </span>
        </div>
      </a>
      <a
        aria-label="Remove note"
        className="remove"
        onClick={[Function]}
        role="link"
        tabIndex="0"
      >
        
      </a>
    </div>
  </div>
  <div
    aria-live="assertive"
    className="reader-only"
    id="notes-assert-container"
    role="alert"
  />
</div>
`;

exports[`renders the Note List 2`] = `
<div
  id="notes"
  role="main"
>
  <div
    className="notes-body"
  >
    <div
      className="note-row"
    >
      <a
        className="note-link"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyUp={[Function]}
        role="link"
        tabIndex="0"
      >
        <span
          className="undefined color-dot"
        >
          <span
            className="reader-only"
          >
             Highlight
          </span>
        </span>
        <div
          className="note-content"
        >
          <div>
            <p
              className="dotdotdot"
            >
              <strong />
            </p>
          </div>
          <div
            className="hide"
          >
            <p
              className="user-note"
            />
          </div>
          <span
            className="note-date"
          >
            <time
              value={undefined}
            >
              Invalid Date
            </time>
             
            <time
              value={undefined}
            >
              Invalid Date
            </time>
          </span>
        </div>
      </a>
      <a
        aria-label="Remove note"
        className="remove"
        onClick={[Function]}
        role="link"
        tabIndex="0"
      >
        
      </a>
    </div>
    <div
      className="note-row"
    >
      <a
        className="note-link"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyUp={[Function]}
        role="link"
        tabIndex="0"
      >
        <span
          className="undefined color-dot"
        >
          <span
            className="reader-only"
          >
             Highlight
          </span>
        </span>
        <div
          className="note-content"
        >
          <div>
            <p
              className="dotdotdot"
            >
              <strong />
            </p>
          </div>
          <div
            className="hide"
          >
            <p
              className="user-note"
            />
          </div>
          <span
            className="note-date"
          >
            <time
              value={undefined}
            >
              Invalid Date
            </time>
             
            <time
              value={undefined}
            >
              Invalid Date
            </time>
          </span>
        </div>
      </a>
      <a
        aria-label="Remove note"
        className="remove"
        onClick={[Function]}
        role="link"
        tabIndex="0"
      >
        
      </a>
    </div>
    <div
      className="note-row"
    >
      <a
        className="note-link"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyUp={[Function]}
        role="link"
        tabIndex="0"
      >
        <span
          className="undefined color-dot"
        >
          <span
            className="reader-only"
          >
             Highlight
          </span>
        </span>
        <div
          className="note-content"
        >
          <div>
            <p
              className="dotdotdot"
            >
              <strong />
            </p>
          </div>
          <div
            className="hide"
          >
            <p
              className="user-note"
            />
          </div>
          <span
            className="note-date"
          >
            <time
              value={undefined}
            >
              Invalid Date
            </time>
             
            <time
              value={undefined}
            >
              Invalid Date
            </time>
          </span>
        </div>
      </a>
      <a
        aria-label="Remove note"
        className="remove"
        onClick={[Function]}
        role="link"
        tabIndex="0"
      >
        
      </a>
    </div>
  </div>
  <div
    aria-live="assertive"
    className="reader-only"
    id="notes-assert-container"
    role="alert"
  />
</div>
`;
